packages/dds/tree/docs/wip/cell-model.md

```diff
@@ -0,0 +1,183 @@
+# The Cell Model of Collaborative Editing
+
+This document presents a conceptual model that provides a foundation upon which the semantics of SharedTree editing can be understood.
+
+The model itself is not tied to SharedTree or SharedTree's data model.
+It is instead concerned with providing an abstraction for what it means to edit a document.
+
+The model only concerns itself with the editing of a single field.
+More complex structures (notably, trees) can be constructed by allowing the contents of the field to represent more fields.
+
+## Motivation
+
+We present here an example that portrays the challenges encountered by editing models based on insertion and removal.
+
+Consider a collaborative editing session between three participants (all acting concurrently), over a field populated with elements [A, B, C, D], where the edits are sequenced as follows:
+
+1. User 1: remove B, C (local state: [A, D])
+
+2. User 2: insert Y after C (local state: [A, B, C, Y, D])
+
+3. User 3: insert X after B (local state: [A, B, X, C, D])
+
+4. User 1: undo the removal of A, B, C (local state: [A, B, C])
```

Comment:
I think you mean "undo the removal of B, C (local state: [A, B, C, D])"

Reply:
Similarly on line 35.

Reply:
Thanks!